domain/relation/service/relation.go

```diff
@@ -42,9 +42,26 @@ func NewService(
 	}
 }
 
-// AddRelation adds a relation between the two provided endpoints.
-func (s *Service) AddRelation(ctx context.Context, eps []relation.Endpoint) error {
-	return errors.NotImplemented
+// AddRelation takes two endpoint identifiers of the form
+// <application>[:<endpoint>]. The identifiers will be used to infer two
+// endpoint between applications on the model. A new relation will be created
+// between these endpoints and the details of the endpoint returned.
+//
+// If the identifiers do not uniquely specify a relation, an error will be
+// returned.
+func (s *Service) AddRelation(ctx context.Context, ep1, ep2 string) (relation.Endpoint, relation.Endpoint, error) {
+	return relation.Endpoint{}, relation.Endpoint{}, errors.NotImplemented
+}
+
+// AddPeerRelation takes an identifier of the form <application>[:<endpoint>].
+// The identifier will be used to infer a peer endpoint on an application on the
+// model. A new peer relation will be created on this endpoint and the endpoint
+// returned.
+//
+// If the identifier does not specify a peer relation, an error will be
+// returned.
+func (s *Service) AddPeerRelation(ctx context.Context, ep string) (relation.Endpoint, error) {
```

Comment:
issue: delete me. Peer relations are added when an application is created and updated when it's refreshed. Not by direct service method.